worker/upgrader/upgrader_test.go

```diff
@@ -427,7 +427,7 @@ func (s *UpgraderSuite) TestUpgraderForbidsDowngradingToMajorVersion(c *gc.C) {
 	_, err = agenttools.ReadTools(s.dataDir, downgradeTools.Version)
 	// TODO: ReadTools *should* be returning some form of
 	// errors.NotFound, however, it just passes back a fmt.Errorf so
-	// we live with it c.Assert(err, jc.Satisfies, errors.IsNotFound)
+	// we live with it c.Assert(err, jc.ErrorIs, errors.NotFound)
```

Comment:
Can we do a drive by please and fix this TODO. Otherwise it will collect dust like a fine wine but never actually taste good :)

Reply:
Happy to help with the implementation. I had a look and it's a really quick 2 line change in the func.

Just need to use %w in the fmt.Errorf and can also use `errors.Hide` if we want to keep the error string out of the error.

Reply:
I'd rather keep a functionality change like this out of this PR really